airflow/www/views.py

```diff
@@ -2329,28 +2411,29 @@ def process_form(self, form, is_created):
             form.extra.data = json.dumps(extra)
 
     def prefill_form(self, form, pk):
+        """Prefill the form."""
         try:
-            d = json.loads(form.data.get('extra', '{}'))
+            extra_dictionary = json.loads(form.data.get('extra', '{}'))
         except JSONDecodeError:
-            d = {}
+            extra_dictionary = {}
 
-        if not hasattr(d, 'get'):
-            logging.warning('extra field for {} is not iterable'.format(
-                form.data.get('conn_id', '<unknown>')))
+        if not hasattr(extra_dictionary, 'get'):
```

Comment:
Hm, should this be more explicit?
```suggestion
        if not isinstance(extra_dictionary, dict):
```
I may be wrong but only `dict` type (from json.loads types)  has `get` method?

Reply:
Could be - but I prefered to have minimal number of actual logic change in this PR and focus on refactoring removing pylint warnings without impacting the logic too much as it would be a bit risky. I'd prefer to leave it as-is. There might be some cases where this object is different than a dict, so i prefer to leave it as is. I think the right time to change it is when we add mypy typing - then we will be more certain of the types of each file, but IMHO that should be a different change. I would love to keep it only focused on naming and other stuff which is pylint-related.

Reply:
> There might be some cases where this object is different than a dict, so i prefer to leave it as is

Any examples? I personally think that we should make pylint changes by fxing code not by adding disables. At least I think that was an assumption made some time ago. I am afraid that if we add disables everywhere then no one ever will fix the code.

Reply:
Actually it was even worse - because the error message was wrong (it was telling about the extra not being iterable.

Reply:
Fixed.